igvm/hypervisor.py

```diff
@@ -570,12 +571,21 @@ def migrate_vm(
                     vm.set_state('maintenance', transaction=transaction)
 
                     if vm.is_running():
-                        vm.shutdown(transaction)
+                        if no_shutdown:
+                            log.info('Please shut down the VM manually now')
+                            vm.wait_for_running(running=False, timeout=86400)
+                        else:
+                            vm.shutdown(transaction)
```

Comment:
Shouldn't we once more wait for wait_for_sync() right after the shutdown, especially now that the shutdown can be up to one day after the wait_for_sync above?

Reply:
Not really. `wait_for_sync()` waits for initial "raid rebuild". After that is done, disk is in constant replication which is only torn down after VM is already stopped after `with` in line 567.

Reply:
Looking at the context manager start() and the called stop() in it's finally branch, I think there is also no logic to wait for the disk to be in sync after the shutdown. It seems like the stop branch just forcefully stops DRBD either way. Or am I missing something?

Reply:
I am not 100% sure. I would have to investigate it and see what stop() really does. If there is a real bug, then it is in stop(), not here.

Reply:
I was unable to find if `drbdadm down` ensures data consistency, but I changed protocol to C and re-enabled flushing, so any IO operation must be written down on both hosts before IO operation is finished.